holidays/countries/malaysia.py

```diff
@@ -408,230 +392,390 @@ def _populate(self, year):
-            self[date(year, JUL, 17)] = "Birthday of The Raja of Perlis"
+            if year >= 2018:
+                hol_date = date(year, JUL, 17)
+            elif year >= 2000:
+                hol_date = date(year, MAY, 17)
+            self[hol_date] = "Birthday of The Raja of Perlis"
 
         elif self.subdiv == "SGR":
             self[date(year, DEC, 11)] = "Birthday of The Sultan of Selangor"
 
-        elif self.subdiv == "SBH":
-            first_sat_oct = rrule(
-                MONTHLY,
-                dtstart=date(year, OCT, 1),
-                count=1,
-                bysetpos=1,
-                byweekday=SA,
-            )[0]
-            self[first_sat_oct] = "Birthday of the Governor of Sabah"
-            if year > 2018:
-                self[date(year, DEC, 24)] = "Christmas Eve"
-
         elif self.subdiv == "TRG":
-            self[
-                date(year, MAR, 4)
-            ] = "Anniversary of the Installation of the Sultan of Terengganu"
-            self[date(year, APR, 26)] = "Birthday of the Sultan of Terengganu"
+            if year >= 2000:
+                self[date(year, MAR, 4)] = (
+                    "Anniversary of the Installation "
+                    "of the Sultan of Terengganu"
+                )
+                self[
+                    date(year, APR, 26)
+                ] = "Birthday of the Sultan of Terengganu"
 
-    def my_islamic_to_gre(self, year: int, month: int, day: int):
-        """
-        Malaysia seems to have a slightly different Hijri calendar. This
-        function returns the adjusted date.
+        # Check for holidays that fall on a Sunday and
+        # implement Section 3 of Malaysian Holidays Act:
+        # "if any day specified in the Schedule falls on
+        # Sunday then the day following shall be a public
+        # holiday and if such day is already a public holiday,
+        # then the day following shall be a public holiday"
+        # In Johor and Kedah it's Friday -> Sunday,
+        # in Kelantan and Terengganu it's Saturday -> Sunday
+        for (hol_date, hol_name) in list(self.items()):
+            if hol_date.year == year:
+                if hol_date.weekday() == FRI and self.subdiv in ("JHR", "KDH"):
+                    in_lieu_date = hol_date + rd(days=+2)
+                elif hol_date.weekday() == SAT and self.subdiv in (
+                    "KTN",
+                    "TRG",
+                ):
+                    in_lieu_date = hol_date + rd(days=+1)
+                elif hol_date.weekday() == SUN and self.subdiv not in (
+                    "JHR",
+                    "KDH",
+                    "KTN",
+                    "TRG",
+                ):
+                    in_lieu_date = hol_date + rd(days=+1)
+                else:
+                    continue
+                while in_lieu_date in self:
+                    in_lieu_date += rd(days=+1)
+                self[in_lieu_date] = hol_name + " [In lieu]"
```

Comment:
```suggestion
        for hol_date, hol_name in list(self.items()):
            if hol_date.year != year:
                continue

            in_lieu_date = None
            if hol_date.weekday() == FRI and self.subdiv in ("JHR", "KDH"):
                in_lieu_date = hol_date + rd(days=+2)
            elif hol_date.weekday() == SAT and self.subdiv in (
                "KTN",
                "TRG",
            ):
                in_lieu_date = hol_date + rd(days=+1)
            elif hol_date.weekday() == SUN and self.subdiv not in (
                "JHR",
                "KDH",
                "KTN",
                "TRG",
            ):
                in_lieu_date = hol_date + rd(days=+1)

            if not in_lieu_date:
                continue

            while in_lieu_date in self:
                in_lieu_date += rd(days=+1)
            self[in_lieu_date] = hol_name + " [In lieu]"
```

Reply:
Could you try this code for 100% coverage?

Reply:
Thank you, it's worked!